lib/webpage_cache_service.rb

```diff
@@ -0,0 +1,133 @@
+class WebpageCacheService
+  DEFAULT_HEADERS = {
+    user_agent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:57.0) Gecko/20100101 Firefox/57.0",
+    accept_language: "en-US,en;q=0.5",
+    accept: "application/html;q=0.9,*/*;q=0.8; charset=utf-8"
+  }.freeze
+
+  ROOT = Rails.root.join("storage", "webpage_caches").freeze
```

Comment:
how much disk space can I have? I bet I could make transientbug cache turing complete...

Reply:
Probably :P Limited space though. Next iteration is going to have a cap on the size of offline caches to help slow down the growth however.